wgsl/index.bs

```diff
@@ -7344,6 +7344,9 @@ Note: The [=attribute/const=] attribute cannot be applied to user-declared funct
 * A [=formal parameter|function parameter=] [=shader-creation error|must=] one the following types:
     * a constructible type
     * a pointer type
+        * It [=shader-creation error|must=] be valid to [=variable declaration|declare a variable=]
+              whose [=reference type=] corresponds to the pointer type,
+              i.e. with the pointer type's address space, store type, and access mode.
```

Comment:
This works for me. You can't declare a dynamically-sized type in any of the address spaces functions accept pointers in, so this addresses my concern in #2268.